src/tls.c

```diff
@@ -269,7 +275,13 @@ static int doPRF(byte* digest, word32 digLen, const byte* secret,word32 secLen,
     byte  md5_result[MAX_PRF_DIG];    /* digLen is real size */
     byte  sha_result[MAX_PRF_DIG];    /* digLen is real size */
 #endif
+#if defined(WOLFSSL_ASYNC_CRYPT) && !defined(WC_ASYNC_NO_HASH)
     DECLARE_VAR(labelSeed, byte, MAX_PRF_LABSEED, heap);
+    if (labelSeed == NULL)
+        return MEMORY_E;
+#else
+    byte labelSeed[MAX_PRF_LABSEED];
+#endif
 
     if (half > MAX_PRF_HALF)
         return BUFFER_E;
```

Comment:
This looks like a memory leak in async case.  Wouldn't labelSeed be on the help?  And now we're returned BUFFER_E.